src/UI/Brush.java

```diff
@@ -0,0 +1,252 @@
+	 * @param endy end(dragged) pt(Y)
+	 * @returns nothing
+	 */
+	public static void drawBrush(
+		Color color,
+		GraphicsContext g,
+		double startx,
+		double starty,
+		double endx,
+		double endy
+	) {
+		g.setStroke(color);
+		g.stroke();
+		g.closePath();
+
+		drawerase.drawCurve(pixels);
+		pixels = new ArrayList<Pixel>();
+	}
+
+	/**
+	 * This is called on mouse release for eraser to send all positions as an arraylist to the processing module
+	 * @param color   any color selected
+	 * @param g   context
+	 * @param startx start pt(X)
+	 * @param starty start pt(Y)
+	 * @param endx end(dragged) pt(X)
+	 * @param endy end(dragged) pt(Y)
+	 * @returns nothing
+	 */
+	public static void drawEraser(
+		Color color,
+		GraphicsContext g,
+		double startx,
+		double starty,
+		double endx,
+		double endy
+	) {
+		g.closePath();
+
+		drawerase.erase(position);
+		position = new ArrayList<Position>();
+	}
+
+	/**
+	 * It is to show the brush effect while mouse drag and collect data to send to processing team
+	 * @param canvas  canvas to do on
+	 * @param color   any color selected
+	 * @param g   context
+	 * @param startx start pt(X)
+	 * @param starty start pt(Y)
+	 * @param endx end(dragged) pt(X)
+	 * @param endy end(dragged) pt(Y)
+	 * @param size brushsize
+	 * @returns nothing
+	 */
+	public static void drawBrushEffect(
+		Canvas canvasB,
+		Color color,
+		GraphicsContext g,
+		double startx, double starty, double endx, double endy,double size
+	) {
+		g.stroke();
+		g.setLineCap(StrokeLineCap.ROUND);
+		g.setLineWidth(size);
+		g.setStroke(color);
+		g.strokeLine(startx,starty,endx,endy);
+
+		double x = 0, y = 0, m = 0, flag = 0;
+
+		/**
+		 * Check if slope is 0 or infinity.
+		 * If either is there flag to a certain value
+		 * m = slope
+		 */
+		if(startx == endx) {
+			flag = 1;
+		}
+		else if(endy == starty) {
+			flag = 2;
+		}
+		else {
+			flag = 0;
+			m = (endy - starty) / (endx - startx);
+		}
+
+		/**
+		 * Intensity in terms of RGB format
+		 */
+		Intensity i = new Intensity((int) color.getRed(), (int) color.getGreen(), (int) color.getBlue());
+
+		/**
+		 * The calculation of pixels
+		 * Outer loop is for every 2 distance for line drawn from (startx , starty) to (endx , endy)
+		 * Inner loop is half the size on both sides of line with distance 1.
+		 * j and k are local loop variables.
+		 */
+		for (
+			int j = 0;
+			j <= (int)(Math.sqrt((endy - starty) * (endy - starty) + (endx - startx) * (endx - startx)));
+			j+=2
+		) {
+			for (int k = (int)(-1 * size/2); k <= (int)(size/2); k+=1) {
+				if (flag == 0) {
+					x = startx + Math.sqrt((j*j)/(m * m + 1)) + ( Math.abs(m * k) * ( Math.sqrt(1/(1 + m * m)) ));
+					y = starty + ( Math.abs(m) * Math.sqrt((j*j)/(m * m + 1))) - ( (1/m) * Math.abs(m * k) * Math.sqrt(1/(1 + m * m)));
+				}
+				else if (flag == 1) {
+					x = startx + k;
+					y = starty + j;
+				}
+				else if (flag == 1) {
+					x = startx + j;
+					y = starty + k;
+				}
```

Comment:
Blunder! same condition within nested if-else.

Reply:
Changes Corrected. 2nd one changes to flag==2

Reply:
resolved